File: 백준/Gold/6597. 트리 복구/트리 복구.py
import sys

def solution():

    def postorder(order):
        if not order:
            return
        nonlocal idx, res
        temp = preorder[idx]
        left, right = order.split(preorder[idx])
        idx += 1
        postorder(left)
        postorder(right)
        res += temp

    cases = sys.stdin.readlines()
    for case in cases:
        preorder, inorder = case.split()
        idx = 0
        res = ''
        postorder(inorder)
        print(res)

solution()

File: 백준/Gold/6597. 트리 복구/트리 복구.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

function solution() {
  let answer;
  let preorder;
  let inorder;
  let idx;
    
  function postorder(order) {
      if (order === '') return;
      const temp = preorder[idx];
      const [left, right] = order.split(preorder[idx]);
      idx += 1;
      postorder(left);
      postorder(right);
      res += temp;
  }
  
  for (let i=0; i<input.length-1; i++) {
      [preorder, inorder] = input[i].split(' ');
      idx = 0;
      res = '';
      postorder(inorder);
      console.log(res);
  }
}

solution();